Lib/test/test_asyncio/test_taskgroups.py

```diff
@@ -0,0 +1,698 @@
+                t1 = g.create_task(foo1())
+                t2 = g.create_task(foo2())
+                1 / 0
+
+        try:
+            await runner()
+        except ExceptionGroup as t:
+            self.assertEqual(get_error_types(t), {ZeroDivisionError})
+        else:
+            self.fail('ExceptionGroup was not raised')
+
+        self.assertTrue(t1.cancelled())
+        self.assertTrue(t2.cancelled())
+
+    async def test_taskgroup_11(self):
+
+        async def foo():
+            await asyncio.sleep(0.1)
+            1 / 0
+
+        async def runner():
+            async with taskgroups.TaskGroup():
+                async with taskgroups.TaskGroup() as g2:
+                    for _ in range(5):
+                        g2.create_task(foo())
+
+                    try:
+                        await asyncio.sleep(10)
+                    except asyncio.CancelledError:
+                        raise
+
+        r = self.loop.create_task(runner())
+        await asyncio.sleep(0.1)
+
+        self.assertFalse(r.done())
+        r.cancel()
+        with self.assertRaises(asyncio.CancelledError):
+            await r
+
+    async def test_taskgroup_12(self):
+
+        async def foo():
+            await asyncio.sleep(0.1)
+            1 / 0
+
+        async def runner():
+            async with taskgroups.TaskGroup() as g1:
+                g1.create_task(asyncio.sleep(10))
+
+                async with taskgroups.TaskGroup() as g2:
+                    for _ in range(5):
+                        g2.create_task(foo())
+
+                    try:
+                        await asyncio.sleep(10)
+                    except asyncio.CancelledError:
+                        raise
+
+        r = self.loop.create_task(runner())
+        await asyncio.sleep(0.1)
+
+        self.assertFalse(r.done())
+        r.cancel()
+        with self.assertRaises(asyncio.CancelledError):
+            await r
+
+    async def test_taskgroup_13(self):
+
+        async def crash_after(t):
+            await asyncio.sleep(t)
+            raise ValueError(t)
+
+        async def runner():
+            async with taskgroups.TaskGroup(name='g1') as g1:
+                g1.create_task(crash_after(0.1))
+
+                async with taskgroups.TaskGroup(name='g2') as g2:
+                    g2.create_task(crash_after(0.2))
+
+        r = self.loop.create_task(runner())
+        with self.assertRaises(ExceptionGroup) as cm:
+            await r
+
+        self.assertEqual(get_error_types(cm.exception), {ValueError})
+
+    async def test_taskgroup_14(self):
+
+        async def crash_after(t):
+            await asyncio.sleep(t)
+            raise ValueError(t)
+
+        async def runner():
+            async with taskgroups.TaskGroup(name='g1') as g1:
+                g1.create_task(crash_after(0.2))
+
+                async with taskgroups.TaskGroup(name='g2') as g2:
+                    g2.create_task(crash_after(0.1))
```

Comment:
This code seems not robust; it can fail on a slow machine.
Perhaps waiting for `asyncio.Event` (or future) is better than constant small sleeps (but sleeps are better readable, sure).

Reply:
Let me fix that by making the longer sleep longer -- that sleep will be interrupted so it doesn't matter how long it is. Let me know if you think that's not the right way to fix this test.